frame/system/src/lib.rs

```diff
@@ -537,8 +541,10 @@ decl_module! {
 		/// - 1 storage write (codec `O(C)`).
 		/// - 1 call to `can_set_code`: `O(S)` (calls `sp_io::misc::runtime_version` which is expensive).
 		/// - 1 event.
+		/// The weight of this function is dependent on the runtime, but generally this is very expensive.
+		/// We will treat this as a full block.
 		/// # </weight>
-		#[weight = (200_000_000, DispatchClass::Operational)]
+		#[weight = (crate::Module::<T>::max_extrinsic_weight(DispatchClass::Operational), DispatchClass::Operational)]
```

Comment:
Should the DispatchClass of `set_code` be `Mandatory`?

Also please someone double check this `max_extrinsic_weight` logic

Reply:
This doesnt work, max_extrinsic_weight conflicts with timestamp stuff i believe and exhausts resources.